### PR TITLE
Centralize name generation in one place

### PR DESCRIPTION
**Pull request by chamons**
- Both to get more code out of objcgenerator*.cs and pave the way to possible future caching

### CODE REVIEW

**Comment by chamons** [comment]
Hmm. I swear that built locally....Fixing

**Comment by chamons** [comment]
I only ran tests in tests/objc-cli locally. My bad :(

File: objcgen/NameGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Type = IKVM.Reflection.Type;
+
+namespace ObjC
+{
```
Comment (dalexsoto) [inline objcgen/NameGenerator.cs]: Minor: Mono Coding Guidelines 😄

File: objcgen/NameGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	public static class NameGenerator
+	{
```
Comment (dalexsoto) [inline objcgen/NameGenerator.cs]: Minor: Mono Coding Guidelines

File: objcgen/NameGenerator.cs (hunk 1 above)
Comment (chamons) [inline objcgen/NameGenerator.cs]: I swear XS is "undoing" these when I'm not looking...

**Pull request merged**